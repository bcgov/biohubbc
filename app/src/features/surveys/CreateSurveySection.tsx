import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { useBiohubApi } from 'hooks/useBioHubApi';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';

export interface IProjectSurveySectionProps {
  title: string;
  summary: string;
  component: any;
}

/**
 * Page to display user management data/functionality.
 *
 * @return {*}
 */
const CreateSurveySection: React.FC<IProjectSurveySectionProps> = (props) => {
  const { title, summary, component } = props;

  const urlParams = useParams();

  const biohubApi = useBiohubApi();

  const [isLoadingCodes, setIsLoadingCodes] = useState(false);
  const [codes, setCodes] = useState<IGetAllCodeSetsResponse>();

  useEffect(() => {
    const getCodes = async () => {
      const codesResponse = await biohubApi.codes.getAllCodeSets();

      if (!codesResponse) {
        // TODO error handling/messaging
        return;
      }

      setCodes(codesResponse);
    };

    if (!isLoadingCodes && !codes) {
      getCodes();
      setIsLoadingCodes(true);
    }
  }, [urlParams, biohubApi.codes, isLoadingCodes, codes]);

  return (
    <>
      <Box width="100%" display="flex" flexWrap="nowrap">
        <Box width="40%" p={3}>
          <Box>
            <Typography variant="h2">{title}</Typography>
          </Box>
          <Box>
            <Typography variant="body2">{summary}</Typography>
          </Box>
        </Box>
        <Box width="60%" p={3}>
          {component}
        </Box>
      </Box>
    </>
  );
};

export default CreateSurveySection;
