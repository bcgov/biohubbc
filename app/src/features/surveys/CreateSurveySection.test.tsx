import { render } from '@testing-library/react';
import CreateSurveySection from './CreateSurveySection';
import React from 'react';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

describe('Create Survey Section', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <CreateSurveySection
        title="Section title"
        summary="Section summary"
        component={
          <Box>
            <Typography>Hey this is my content</Typography>
          </Box>
        }
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
