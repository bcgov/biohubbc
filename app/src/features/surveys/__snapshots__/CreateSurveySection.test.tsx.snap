// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Survey Section renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Section title
        </h2>
      </div>
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Section summary
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hey this is my content
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
